packages/editor/src/store/effects/posts.js

```diff
@@ -201,27 +202,47 @@ export const requestPostUpdateSuccess = ( action, store ) => {
 		noticeMessage = null;
 	} else if ( isPublished && ! willPublish ) {
 		// If undoing publish status, show specific notice
-		noticeMessage = __( 'Post reverted to draft.' );
+		noticeMessage = postTypeSlug === 'page' ? __( 'Page reverted to draft.' ) : __( 'Post reverted to draft.' );
 		shouldShowLink = false;
 	} else if ( ! isPublished && willPublish ) {
 		// If publishing or scheduling a post, show the corresponding
 		// publish message
-		noticeMessage = {
-			publish: __( 'Post published!' ),
-			private: __( 'Post published privately!' ),
-			future: __( 'Post scheduled!' ),
-		}[ post.status ];
+		const noticeMessagePublished = {
+			page: {
+				publish: __( 'Page published!' ),
+				private: __( 'Page published privately!' ),
+				future: __( 'Page scheduled!' ),
+			},
+			post: {
+				publish: __( 'Post published!' ),
+				private: __( 'Post published privately!' ),
+				future: __( 'Post scheduled!' ),
+			},
+		};
+		noticeMessage = noticeMessagePublished[ postTypeSlug ][ post.status ] || noticeMessagePublished.post[ post.status ];
 	} else {
 		// Generic fallback notice
-		noticeMessage = __( 'Post updated!' );
+		const noticeMessageUpdated = {
+			page: {
+				publish: __( 'Page updated!' ),
+				private: __( 'Page updated privately!' ),
+				future: __( 'Page scheduled!' ),
+			},
+			post: {
+				publish: __( 'Post updated!' ),
+				private: __( 'Post updated privately!' ),
+				future: __( 'Post scheduled!' ),
+			},
+		};
+		noticeMessage = noticeMessageUpdated[ postTypeSlug ][ post.status ] || noticeMessageUpdated.post[ post.status ];
```

Comment:
Thus you mean instead of

```
noticeMessage = noticeMessageUpdated[ postTypeSlug ][ post.status ] || noticeMessageUpdated.post[ post.status ];
```

I could simply use

```
noticeMessage = noticeMessageUpdated.post[ post.status ];
```

Reply:
No. What I meant is more like

```
noticeMessage = noticeMessageUpdated[ postTypeSlug ][ post.status ] || ( isPostTypeHierarchical ? noticeMessageUpdated.page[ post.status ] : noticeMessageUpdated.post[ post.status ] )
```

See `get_post_type_labels()` and `_get_custom_object_labels()` in core.